### PR TITLE
sql: New tables added will have columns in the correct order

### PR DESCRIPTION
**Pull request by Fenil-P**
Release note (sql change): Reordered unimplemented vtables columns in pg_catalog and information_schema

### CODE REVIEW

**Review by rafiss — CHANGES_REQUESTED** [review]
looking good! we should also add more to this PR so that existing tables that are entirely empty have their column orders fixed.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75461#-:-MuHxn__D2MO6sK532Ya:bjv1wfd)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Fenil-P and @RichardJCai)

___
*[pkg/sql/testdata/postgres_test_expected_diffs_on_pg_catalog.json, line 7 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/75461#-MuHxTm98qyqPGdDJOf3:-MuHxTmA7rrcvZeVtuuh:b5fag0w) ([raw file](https://github.com/cockroachdb/cockroach/blob/63090d5faae091dc7b8da9af02e007f310252af8/pkg/sql/testdata/postgres_test_expected_diffs_on_pg_catalog.json#L7)):*
> ```JSON
>     "TotalColumns": 1267,
>     "MissingTables": 0,
>     "MissingColumns": 2,
> ```

do you know what the two missing columns are? for some of them, we might be able to add the column with a different type

**Review by rafiss — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75461)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Fenil-P and @RichardJCai)

___
*[pkg/sql/testdata/pg_catalog_tables_from_postgres.json, line 2 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/75461#-MuHyETD4c2v7iY01Szg:-MuHyETD4c2v7iY01Szh:bpft0fe) ([raw file](https://github.com/cockroachdb/cockroach/blob/0bd1537a385beca949a2c7ee3ff2595483ca69f8/pkg/sql/testdata/pg_catalog_tables_from_postgres.json#L2)):*
> ```JSON
> {
>   "version": "13.5",
> ```

ah, testing with postgres 14 would make sense; but let's do that as a separate PR to keep the scope of this one smaller

**Review by Fenil-P — COMMENTED** [review]
Sounds good, working on that next

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75461#-:-MuI1OD31iA1nwuHoKpW:b-2tkpcu)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Fenil-P, @rafiss, and @RichardJCai)

___
*[pkg/sql/testdata/pg_catalog_tables_from_postgres.json, line 2 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/75461#-MuHyETD4c2v7iY01Szg:-MuI1ECDDV_W-xi_gm8h:bfisz9c) ([raw file](https://github.com/cockroachdb/cockroach/blob/0bd1537a385beca949a2c7ee3ff2595483ca69f8/pkg/sql/testdata/pg_catalog_tables_from_postgres.json#L2)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

ah, testing with postgres 14 would make sense; but let's do that as a separate PR to keep the scope of this one smaller
</blockquote></details>

Yup, using 13.5 now and the tests work now

**Comment by RichardJCai** [comment]
The PR body is no longer accurate right? No new tables were added?

```
Release note (sql change): Added missing tables at pg_catalog
New Tables Added:
pg_stats_ext_exprs
pg_stat_progress_copy
pg_backend_memory_contexts
pg_stat_wal
pg_stat_replication_slots
```

File: pkg/sql/pg_metadata_diff.go [hunk 1]
```diff
@@ -86,8 +85,14 @@ type PGMetadataColumnType struct {
 // PGMetadataColumns maps columns names to datatypes.
 type PGMetadataColumns map[string]*PGMetadataColumnType
 
+// PGMetadataTableInfo represents a table with column mapping and insertion order.
+type PGMetadataTableInfo struct {
+	Order   *[]string         `json:"order"`
```
Comment (RichardJCai) [inline pkg/sql/pg_metadata_diff.go:90]: nit: why is this a pointer to a slice? 

Another nit, can we rename order to `ColumnNames` and leave a comment that it must be in sorted order.

**Review by RichardJCai — COMMENTED** [review]
Almost LGTM besides little nits, I only spotchecked some of the tables that were updated.

Can we make a separate logic test file that checks which tables are implemented / unimplemented and make sure that didn't change.

We could make a `crdb_internal` table that iterates through the `pgCatalog` virtual schema and returns whether a table is implemented or not.

See how tables are implemented in `crdb_internal.go` we can add a new table like `crdb_internal.pg_catalog_table_implemented` or something along those lines to do a comparison to make sure we didn't accidentally remove an unimplemented table or go from unimplemented to implemented.

**Review by Fenil-P — COMMENTED** [review]
Resolved, the crdb_internal tests pass so there were no changes to the unimplemented property. No new tables added either

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75461#-:-MuvFebXEH-9n-ym8EFs:bic7m51)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @RichardJCai)

___
*[pkg/sql/testdata/postgres_test_expected_diffs_on_pg_catalog.json, line 7 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/75461#-MuHxTm98qyqPGdDJOf3:-MuvFzQe0SqyczLDx5PY:b5k66g9) ([raw file](https://github.com/cockroachdb/cockroach/blob/63090d5faae091dc7b8da9af02e007f310252af8/pkg/sql/testdata/postgres_test_expected_diffs_on_pg_catalog.json#L7)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

do you know what the two missing columns are? for some of them, we might be able to add the column with a different type
</blockquote></details>

Resolved by using 13.3

**Review by RichardJCai — APPROVED** [review]
Nice LGTM!

**Pull request merged**